### PR TITLE
Small simplifications to Serial.cpp

### PR DESCRIPTION
**Pull request by MikeGitb**
Mainly refactoring `Serial::readStringUntil`. There is certainly more that could be done, but these are some low-hanging fruits.

### CODE REVIEW

File: src/cinder/Serial.cpp [hunk 1]
```diff
@@ -151,23 +151,21 @@ Serial::Impl::~Impl()
...
 }
 
 Serial::Device Serial::findDeviceByNameContains( const std::string &searchString, bool forceRefresh )
 {
-	const std::vector<Serial::Device> &devices = getDevices( forceRefresh );
-	for( std::vector<Serial::Device>::const_iterator deviceIt = devices.begin(); deviceIt != devices.end(); ++deviceIt ) {
-		if( deviceIt->getName().find( searchString ) != std::string::npos )
-			return *deviceIt;
+	for( const auto& device : getDevices() ) {
+		if(device.getName().find( searchString ) != std::string::npos )
```
Comment (drewish) [inline src/cinder/Serial.cpp]: I think you need a space inside the first paren

File: src/cinder/Serial.cpp [hunk 2]
```diff
@@ -151,23 +151,21 @@ Serial::Impl::~Impl()
...
 	return Serial::Device();
 }
 
 Serial::Device Serial::findDeviceByNameContains( const std::string &searchString, bool forceRefresh )
 {
-	const std::vector<Serial::Device> &devices = getDevices( forceRefresh );
-	for( std::vector<Serial::Device>::const_iterator deviceIt = devices.begin(); deviceIt != devices.end(); ++deviceIt ) {
-		if( deviceIt->getName().find( searchString ) != std::string::npos )
-			return *deviceIt;
+	for( const auto& device : getDevices() ) {
```
Comment (drewish) [inline src/cinder/Serial.cpp]: Should forceRefresh still be passed in as a parameter?

File: src/cinder/Serial.cpp [hunk 3]
```diff
@@ -151,23 +151,21 @@ Serial::Impl::~Impl()
 
 Serial::Device Serial::findDeviceByName( const std::string &name, bool forceRefresh )
 {
-	const std::vector<Serial::Device> &devices = getDevices( forceRefresh );
-	for( std::vector<Serial::Device>::const_iterator deviceIt = devices.begin(); deviceIt != devices.end(); ++deviceIt ) {
-		if( deviceIt->getName() == name )
-			return *deviceIt;
+	for( const auto& device : getDevices() ) {
```
Comment (drewish) [inline src/cinder/Serial.cpp]: Did you mean to drop forceRefresh?

Comment (MikeGitb) [inline src/cinder/Serial.cpp]: No, I just overlooked it.

File: src/cinder/Serial.cpp (hunk 2 above)
Comment (MikeGitb) [inline src/cinder/Serial.cpp]: Same here

File: src/cinder/Serial.cpp (hunk 1 above)
Comment (MikeGitb) [inline src/cinder/Serial.cpp]: Yes, I didn't properly check whitespaces

**Comment by MikeGitb** [comment]
@drewish: Thanks for the feedback. I'll correct the PR tomorrow.

**Comment by MikeGitb** [comment]
@drewish: OK, if fixed the issues you mentioned.

**Comment by andrewfb** [comment]
Thanks for this improvement.

**Pull request merged**